File: apcs/cpp/zj-c461.cpp

#include <iostream>
using namespace std;

int main(){
    int a, b, c;
    cin >> a >> b >> c;
    if (a != 0) a = 1;
    if (b != 0) b = 1;

    bool flag = false;
    if ((a && b) == c){
        cout << "AND\n";
        flag = true;
    }

    if ((a || b) == c){
        cout << "OR\n";
        flag = true;
    }

    if ((a != b) == c){
        cout << "XOR\n";
        flag = true;
    }

    if (!flag) cout << "IMPOSSIBLE\n";
    return 0;
}

File: apcs/python/zj-c461.py
#!/usr/bin/env python 

while True:
    try:
        a, b, c = map(int, input().split())
        if a != 0:
            a = 1
        if b != 0:
            b = 1
        if c == 0:
            c = False
        else:
            c = True
        flag = False
        if (a and b) == c:
            print('AND')
            flag = True
        if (a or b) == c:
            print('OR')
            flag = True
        if (a != b) == c:
            print('XOR')
            flag = True
        if not flag:
            print('IMPOSSIBLE')
    except:
        break
